src/particle.cpp

```diff
@@ -198,11 +205,19 @@ void Particle::event_advance()
   double distance = std::min(boundary().distance, collision_distance());
 
   // Advance particle in space and time
-  for (int j = 0; j < n_coord(); ++j) {
-    coord(j).r += distance * coord(j).u;
-  }
+  this->move_distance(distance);
   this->time() += distance / this->speed();
 
+  // Kill particle if its time exceeds the cutoff
+  if (time() > settings::time_cutoff[static_cast<int>(type())]) {
+    double time_dif = time() - settings::time_cutoff[static_cast<int>(type())];
+    time() = settings::time_cutoff[static_cast<int>(type())];
+
+    double push_back_distance = speed() * time_dif;
+    this->move_distance(-push_back_distance);
+    hit_time_boundary() = true;
```

Comment:
Sorry to be picky, but this should definitely just be a local variable. No need to make the `Particle` any bigger in this case. Minimizing the size of `Particle` matters a lot on GPU.

Reply:
yes, good point, will just change that